rules/aip0164/http_body_test.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0164
+
+import (
+	"strings"
+	"testing"
+
+	dpb "github.com/golang/protobuf/protoc-gen-go/descriptor"
+	"github.com/jhump/protoreflect/desc/builder"
+	"google.golang.org/genproto/googleapis/api/annotations"
+	"google.golang.org/protobuf/proto"
+)
+
+func TestHttpBody(t *testing.T) {
+	tests := []struct {
+		testName   string
+		body       string
+		methodName string
+		msg        string
+	}{
+		{"Valid", "*", "UndeleteBook", ""},
+		{"Invalid", "", "UndeleteBook", "HTTP body"},
+		{"Irrelevant", "", "AcquireBook", ""},
+	}
+
+	for _, test := range tests {
+		t.Run(test.testName, func(t *testing.T) {
+			// Create a MethodOptions with the annotation set.
+			opts := &dpb.MethodOptions{}
```

Comment:
I am surprised you went this route and not `testutils.ParseProto3Tmpl`, but it is fine as it is.

Reply:
I just copied it from one of the other tests that was similar. It'd be good at some point to upgrade all the tests to use `ParseProto3Tmpl`.

Reply:
Ah, that makes sense. Yeah, I wrote about 20 or so rules before writing `ParseProto3Tmpl`, which required me to get a change in to the underlying protoreflect library.

Reply:
I really miss `g4 cp` in GitHub's PR UI, to make these changes more obvious.